impl/src/main/java/org/jboss/weld/config/ConfigurationKey.java

```diff
@@ -322,6 +322,12 @@
      */
     UNUSED_BEANS_EXCLUDE_ANNOTATION("org.jboss.weld.bootstrap.unusedBeans.excludeAnnotation", "javax\\.ws\\.rs.*"),
 
+    /**
+     * If true accessing a SessionScope bean will result it in being refreshed in the HTTP Session attributes.
+     */
+    @Description("Upon access refresh SessionScoped and ConversationScoped beans into the HTTP Session.")
+    ALWAYS_REFRESH_SESSION_CONVERSATION_SCOPE("org.jboss.weld.context.alwaysRefreshSessionAndConversationScope", false),
```

Comment:
Hm, the name is always the tricky part. What about the key `ENFORCE_HTTP_SESSION_REPLICATION` and property `org.jboss.weld.context.enforceHttpSessionReplication`? I don't think we should mention SCOPE, CONVERSATION (conversation context is backed by an HTTP session anyway) and REFRESH sounds like the whole context is refreshed which is not necessarily true and also feels costly ;-).
 
Also I think the description should be more clear. Maybe something like:
> If set to `true` then when a contextual reference for a `@SessionScoped` or `@ConversationScoped` bean is obtained from a context backed by an HTTP session the instance is set again using `HttpSession.setAttribute()`. This allows to trigger session replication in some application servers.

Reply:
It's not `ENFORCE` really, as it's still best effort and not entirely guaranteed albeit common enough to look as if it is enforced most of the time.
Also, the replication is something we do not handle directly with this, we just re-store it into session attributes often enough.
I was thinking about `EAGER_SESSION_ATTR_REFRESH`? Or perhaps `REPEATED_SESSION_ATTR_STORE`? WDYT?

Reply:
Otherwise +1 for the suggested change of description

Reply:
> I was thinking about EAGER_SESSION_ATTR_REFRESH? Or perhaps REPEATED_SESSION_ATTR_STORE? WDYT?

-1 but I understand your concerns about `ENFORCE`. I think we shoud keep `HTTP_SESSION` and `REPLICATION` parts because user would understand these. And the third part should be something like _"Weld tries to urge the container to trigger..."_. 

On the other hand, this property will be probably used by advanced users so we could probably reference the replication semantics. AFAIK in JBoss AS/EAP 6 this was called `SET replication trigger` (it's not supported in WildFly/EAP 7).

Reply:
`REPLICATE_HTTP_SESSION_ON_ACCESS ` sounds good but is not very accurate. The problem is that replication usually happens on bean access when some kind of "GET trigger" is used (e.g. the default in WildFly). What about `RESET_HTTP_SESSION_ATTR_ON_BEAN_ACCESS?`

Reply:
`RESET_HTTP_SESSION_ATTR_ON_BEAN_ACCESS?` Sounds good to me.

Reply:
Ok, then pls incorporate all required changes. Thanks ;-)

Reply:
But of course ;)